streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -722,22 +721,20 @@ private void checkIfUnexpectedUserSpecifiedConsumerConfig(final Map<String, Obje
      * except in the case of {@link ConsumerConfig#BOOTSTRAP_SERVERS_CONFIG} where we always use the non-prefixed
      * version as we only support reading/writing from/to the same Kafka Cluster.
      *
-     * @param streamThread the {@link StreamThread} creating a consumer
      * @param groupId      consumer groupId
      * @param clientId     clientId
      * @return Map of the consumer configuration.
      */
-    public Map<String, Object> getConsumerConfigs(final StreamThread streamThread,
-                                                  final String groupId,
+    public Map<String, Object> getConsumerConfigs(final String groupId,
                                                   final String clientId) {
         final Map<String, Object> consumerProps = getCommonConsumerConfigs();
 
         // add client id with stream client id prefix, and group id
+        consumerProps.put(APPLICATION_ID_CONFIG, groupId);
```

Comment:
This is a trade off I made: instead of reading the `Property` in `StreamPartitionAssignor` I changed to reading and parsing as `StreamsConfig`, for which the application.id is required. The motivation is to avoid calling `getProperty` and do casting. But this trade-off is arguable to me.

Reply:
I see. Just wondering, if we could not add `application.id` to the config within `StreamPartitionAssignor` before passing it into `StreamsConfig` ?